### PR TITLE
Add visualization page for tileable graphs

### PR DESCRIPTION
**Pull request by RandomY-2**
## What do these changes do?

The code creates a visualization page for all tasks.

On the task info page on Mars Web, there is a new field "Graph" for each task:

Screen Shot 2021-08-03 at 6 53 41 PM

If the user clicks on "See Visualization", the user will be directed to the visualization page for the task. The visualization page displays a DAG for the task and a section that displays the tileable information for the task:

Screen Shot 2021-08-03 at 6 56 41 PM

if the user clicks on a tileable: 

Screen Shot 2021-08-03 at 6 57 36 PM

### CODE REVIEW

File: mars/services/web/ui/src/task_info/TaskListPage.js [hunk 1]
```diff
@@ -84,6 +86,11 @@ class TaskList extends React.Component {
                             <TableCell>{task['end_time'] ? formatTime(task['end_time']) : 'N/A'}</TableCell>
                             <TableCell>{Math.floor(task['progress'] * 100).toString() + "%"}</TableCell>
                             <TableCell>{this.formatTaskStatus(task)}</TableCell>
+                            <TableCell>
```
Comment (wjsi) [inline mars/services/web/ui/src/task_info/TaskListPage.js:89]: I'd like to add a `Task Detail` page and put task visualization inside it as a tab page, just like other server objects.

File: mars/services/web/ui/src/task_info/TaskTileableChart.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import { withRouter } from "react-router-dom";
+import { Grid, Paper } from '@material-ui/core';
+import Title from "../Title";
+import * as d3 from "d3";
+import * as dagreD3 from "dagre-d3";
```
Comment (wjsi) [inline mars/services/web/ui/src/task_info/TaskTileableChart.js:6]: To reduce size of `bundle.js`, import components directly.

File: mars/services/web/ui/src/task_info/TaskTileableChart.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+class tileableTileableChart extends React.Component {
```
Comment (wjsi) [inline mars/services/web/ui/src/task_info/TaskTileableChart.js:8]: Use capitalized class names given [Google Style Guide](https://google.github.io/styleguide/javascriptguide.xml).

**Review by wjsi — CHANGES_REQUESTED** [review]

File: mars/services/web/ui/src/task_info/TaskTileableChart.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+                    </Paper>
+                </Grid>
+                <Grid item xs={12} md={6}>
+                    <Paper style={{ padding: 10 }}>
+                        <Grid item xs={12}>
+                            <Title>Tileable Detail: </Title>
+                        </Grid>
+                        <Grid item xs={12}>
+                            {
+                                this.state.selectedTileable
```
Comment (wjsi) [inline mars/services/web/ui/src/task_info/TaskTileableChart.js:134]: It is better to create an independent `TileableDetail` class for further development.

File: mars/services/web/ui/src/task_info/TaskListPage.js (hunk 1 above)
Comment (RandomY-2) [reply mars/services/web/ui/src/task_info/TaskListPage.js]: New Web UI:

a Task Detail tab on left menu:

Screen Shot 2021-08-05 at 8 35 31 PM

when user clicks, a list of all the task ids will be displayed:

Screen Shot 2021-08-05 at 8 36 20 PM

Clicking on one of the ids will bring the user to the visualization page:

Screen Shot 2021-08-05 at 8 36 56 PM

**Comment by wjsi** [comment]
I build and run this PR and find some issues to resolve.

1. It seems that the dashboard page is missing. Please fix it.
2. I did not mean creating a separate `Task Detail` link in the left navigation pane, that looks a little weird. Creating a link here is better for me as every task belongs to a session.
image
3. The url of task (or graph) page should be /session/<session_id>/task/<task_id>.
4. The direction of DAG shall be reversed. For a simple statement

    ```python
    mt.ones((100, 100), chunk_size=50) * 2 + 1
    ```

    the DAG should be TensorOnes -> TensorMultiply -> TensorAdd

5. Try reduce the length of text displayed on nodes. Operand type alone might be sufficient as nobody really care about node keys.

File: mars/services/web/ui/src/task_visualization/TileableDetail.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                        }
+                    </Grid>
+                </Paper>
+            </Grid>
+        );
+    }
+}
+
+TileableDetail.propTypes = {
+    selectedTileable: PropTypes.shape({
```
Comment (wjsi) [inline mars/services/web/ui/src/task_visualization/TileableDetail.js:45]: Rename to `tileable`.

File: mars/services/web/ui/src/task_visualization/TaskTileableChart.js [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    render() {
+        return (
+            <Grid container spacing={3} >
+                <Grid item xs={12}>
+                    <Title>Task</Title>
+                </Grid>
+                <Grid item xs={12} md={6}>
+                    <Paper>
+                        <Grid item xs={12}>
+                            <svg
```
Comment (wjsi) [inline mars/services/web/ui/src/task_visualization/TaskTileableChart.js:125]: Better to create a separate `TaskTileableDetail` class and make `TaskTileableChart` as a child element.

File: mars/services/web/ui/src/task_visualization/TaskTileableChart.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
+import React from 'react';
```
Comment (wjsi) [inline mars/services/web/ui/src/task_visualization/TaskTileableChart.js:1]: Add copyrightcomments. You may copy these lines from other files.

**Review by wjsi — CHANGES_REQUESTED** [review]
Another two issues:

1. `task_visualization` is part of `task_info`, I suggest these two diractories be merged.
2. Scroll bar appears in `TaskTileableChart` unexpectedly, please fix.

File: mars/services/web/ui/src/task_info/TaskTileableChart.js [hunk 8]
```diff
@@ -0,0 +1,176 @@
+/*
```
Comment (wjsi) [inline mars/services/web/ui/src/task_info/TaskTileableChart.js:1]: Rename to `TaskDetail` and rename `TaskTileableDetail` as `TaskDetail`.

File: mars/services/web/ui/src/PageRouter.js [hunk 9]
```diff
@@ -27,6 +27,9 @@ import SupervisorDetailPage from './node_info/SupervisorDetailPage';
 import WorkerDetailPage from './node_info/WorkerDetailPage';
 import SessionListPage from './SessionListPage';
 import TaskListPage from './task_info/TaskListPage';
+const TaskTileableChart = lazy(() => {
```
Comment (wjsi) [inline mars/services/web/ui/src/PageRouter.js:30]: `TaskDetail`?

**Review by wjsi — COMMENTED** [review]
Github API seems down last time. Please re-commit after renaming.

**Review by wjsi — APPROVED** [review]
LGTM

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**